petab/sbml.py

```diff
@@ -481,3 +485,119 @@ def load_sbml_from_file(
 
     return sbml_reader, sbml_document, sbml_model
+
+
+def get_model_for_condition(
+        petab_problem: "petab.Problem",
+        sim_condition_id: str = None,
+        preeq_condition_id: Optional[str] = None,
+) -> Tuple[libsbml.SBMLDocument, libsbml.Model]:
+    """Create an SBML model for the given condition.
+
+    Creates a copy of the model and updates parameters according to the PEtab
+    files. Estimated parameters are set to their ``nominalValue``.
+    Observables defined in the observables table are not added to the model.
+
+    :param petab_problem: PEtab problem
+    :param sim_condition_id: Simulation conditionId for which to generate a
+        model
+    :param preeq_condition_id: Preequilibration conditionId of the settings
+        for which to generate a model. This is only used to determine the
+        relevant output parameter overrides. Preequilibration is not encoded
+        in the resulting model.
+    :return: The generated SBML document, and SBML model
+    """
+    condition_dict = {petab.SIMULATION_CONDITION_ID: sim_condition_id}
+    if preeq_condition_id:
+        condition_dict[petab.PREEQUILIBRATION_CONDITION_ID] = \
+            preeq_condition_id
+    cur_measurement_df = petab.measurements.get_rows_for_condition(
+        measurement_df=petab_problem.measurement_df,
+        condition=condition_dict,
+    )
+    parameter_map, scale_map = \
+        petab.parameter_mapping.get_parameter_mapping_for_condition(
+            condition_id=sim_condition_id,
+            is_preeq=False,
+            cur_measurement_df=cur_measurement_df,
+            sbml_model=petab_problem.sbml_model,
+            condition_df=petab_problem.condition_df,
+            parameter_df=petab_problem.parameter_df,
+            warn_unmapped=True,
+            scaled_parameters=False,
+            fill_fixed_parameters=True,
+            # will only become problematic once the observable and noise terms
+            #  are added to the model
+            allow_timepoint_specific_numeric_noise_parameters=True,
+        )
+    # create a copy of the model
+    sbml_doc = petab_problem.sbml_model.getSBMLDocument().clone()
+    sbml_model = sbml_doc.getModel()
+
+    # fill in parameters
+    def get_param_value(parameter_id: str):
+        """Parameter value from mapping or nominal value"""
+        mapped_value = parameter_map.get(parameter_id)
+        if not isinstance(mapped_value, str):
+            return mapped_value
+
+        # estimated parameter, look up in nominal parameters
+        return petab_problem.parameter_df.loc[mapped_value,
+                                              petab.NOMINAL_VALUE]
+
+    for parameter in sbml_model.getListOfParameters():
+        new_value = get_param_value(parameter.getId())
+        if new_value:
+            parameter.setValue(new_value)
+            # remove rules that would override that value
+            sbml_model.removeRuleByVariable(parameter.getId())
```

Comment:
```suggestion
            if sbml_model.removeRuleByVariable(parameter.getId()) is not None:
                warnings.warn(f'An SBML rule was removed to set the component {parameter.getId()} to a constant value.')
```
Since I guess this may more often be an error than intended behavior.